doc/acl/egress_outer_dscp_change_table.md

```diff
@@ -0,0 +1,505 @@
+
+
+#### Option-B: Augment existing L3/L3V6 tables to include support for metadata action type.
+
+This approach is similar to Option A with the exception that instead of creating new Ingress tables such as MARK_META/V6, the existing L3 and L3V6 tables would be augmented to include SET_METADATA action field. This would still require the creation of an EGR_SET_DSCP table. This has the advantages of Option A with less ingress stage TCAM utilization.
+
+##### User accessible tables
+    /*
+     * L3 table augmentation
+     * |----------------------------------------------------|
+     * |                        |              |            |
+     * |    Match Type          |   Actions    |    Bind    |
+     * |                        |              |   points   |
+     * |----------------------------------------------------|
+     * | No additionnal Match | SET_METADATA*|              |
+     * |----------------------------------------------------|
+     
+     * L3V6 Table augmentation
+     * |----------------------------------------------------|
+     * |                        |              |            |
+     * |    Match Type          |   Actions    |    Bind    |
+     * |                        |              |   points   |
+     * |----------------------------------------------------|
+     * | No additionnal Match | SET_METADATA*|              |
+     * |----------------------------------------------------|
+
+     * EGR_SET_DSCP
+     * |----------------------------------------------------|
+     * |                        |              |            |
+     * |    Match Type          |   Actions    |    Bind    |
+     * |                        |              |   points   |
+     * |----------------------------------------------------|
+     * |  MATCH_METADATA        | SET_DSCP     |    PORT    |
+     * |                        |              |    LAG     |
+     * |----------------------------------------------------|
+     */
+
+
+##### Pros
+- Ease of Implementation: The changes to the Orchagent would be minimal.
+- The TCAM utilization would be minimal with the introduction of just one egress table. This table would be associated with all the ports excluding loopback and internal ports.
+
+##### Cons
+Apart from the cons mentioned in Option A, this approach may not work on all platforms as some platforms use the same TCAM entry space for actions. This can result in the increase of L3 and L3V6 table footprint. Such a change would break backward compatibility.
+Due to this reason, this option may not be viable at all.
+
+
+#### Option-C: Create a dummy table type which translates into MARK_META/V6 and EGR_SET_DSCP as in the approach mentioned above. Manage the metadata internally.
```

Comment:
If we go with Option-C, does that mean we can't create this egress table standalone?

Reply:
Thats correct. The EGR_SET_DSCP table in this design has one match(META_DATA) and one action(SET_DSCP). I am not sure if there is any other use for it currently.
if in future we feel the need we cna expose that table as well and enhance it with more match attributes and actions.